### PR TITLE
feat(dal): layout repository

### PR DESCRIPTION
**Pull request by p-fernandez**
### What change does this PR introduce?

Adding Layout repository. Also adding some enums and interfaces to unify code across the stack.

### Why was this change needed?

First steps of Layout feature.

### CODE REVIEW

**Comment by SebastianStehle** [comment]
I am just curious...what are layouts? Are you planning to make your task manager public?

File: libs/dal/src/repositories/layout/layout.schema.ts [hunk 1]
```diff
@@ -0,0 +1,54 @@
+import * as mongoose from 'mongoose';
+import { Schema } from 'mongoose';
+
+import { LayoutEntity } from './layout.entity';
+
```
Comment (ainouzgali) [inline libs/dal/src/repositories/layout/layout.schema.ts:5]: There is a mongoose plugin we've been using in some other schemas for the soft delete. You could take a look at feed or integration for example :)

import * as mongooseDelete from 'mongoose-delete';

File: libs/dal/src/repositories/layout/layout.repository.ts [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+      {
+        _id,
+        _environmentId,
+        _organizationId,
+      },
+      {
+        isDeleted: true,
+      }
+    );
+  }
```
Comment (ainouzgali) [inline libs/dal/src/repositories/layout/layout.repository.ts:48]: The plugin also allows here to to search for a record and then just do this.layout.delete

**Review by ainouzgali — APPROVED** [review]
Looks good @p-fernandez!. Take a look at the mongoose delete plugin, its saves some trouble :) 
And I loved the enum refactor😍
Also, Ive created a branch for the layout feature, I thought it might be nice to have one for us so we can feel free to merge things that are not completely ready. called: [feat-layouts](https://github.com/novuhq/novu/tree/feat-layouts)

**Comment by p-fernandez** [comment]
> I am just curious...what are layouts? Are you planning to make your task manager public?

No, it is related to the templates of email notifications, to be able to manage different templates based on layouts created by the user. This feature are basic pieces for a future more complex functionality.

**Pull request merged**